src/LaravelSamlIdpServiceProvider.php

```diff
@@ -109,7 +109,7 @@ private function registerServices()
     private function registerEvents()
     {
         $events = $this->app->make(Dispatcher::class);
-        foreach ($this->events as $event => $listeners) {
+        foreach (config('samlidp.events') as $event => $listeners) {
```

Comment:
@abublihi there may be an issue. For anyone upgrading, will their code break if they do nothing. I think it will. Maybe we can default the array of events that are in the current `samlidp.php` config file. Maybe something like this or similar:

```
$default_events = [
        'CodeGreenCreative\SamlIdp\Events\Assertion' => [],
        'Illuminate\Auth\Events\Logout' => [
            'CodeGreenCreative\SamlIdp\Listeners\SamlLogout',
        ],
        'Illuminate\Auth\Events\Authenticated' => [
            'CodeGreenCreative\SamlIdp\Listeners\SamlAuthenticated',
        ],
        'Illuminate\Auth\Events\Login' => [
            'CodeGreenCreative\SamlIdp\Listeners\SamlLogin',
        ],
];

foreach (config('samlidp.events', $default_events) as $event => $listeners) {
```